sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -107,6 +107,11 @@ public void addCredits(int credits) {
         }
     }
 
+    @Override
+    public void addCreditsInstantly(int credits) {
+        receiver.flow(credits);
```

Comment:
Most of the calls into Proton-j Reactor is done via ReactorDispatcher because of thread safety,  
As commented here ..
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorDispatcher.java#L22
Would this be okay to call this api directly ? @srnagar  Would this be okay here ?

Reply:
This needs to be tested for thread-safety. If proton-j doesn't support adding credits in a thread-safe manner we might add incorrect number of credits to the link and can potentially result in data loss if the SB mode is RECEIVE_AND_DELETE

Reply:
I simply put it into a synchronized block. The performance overhead should be minimal.

Reply:
Changed back to addCredits()